client/src/pages/admin/tasks.tsx

```diff
@@ -24,21 +22,26 @@ import {
 
 const { Content } = Layout;
 type Props = { session: Session; courses: Course[] };
-type ModalData = (Partial<Omit<Task, 'attributes'>> & { attributes?: string }) | null;
-const service = new TaskService();
+type ModalData = (Partial<Omit<TaskDto, 'attributes'>> & { attributes?: string }) | null;
+
+const tasksApi = new TasksApi();
+const criteriaApi = new TasksCriteriaApi();
 const disciplinesApi = new DisciplinesApi();
 
 function Page(props: Props) {
-  const [data, setData] = useState([] as Task[]);
+  const [data, setData] = useState([] as TaskDto[]);
   const [disciplines, setDisciplines] = useState<DisciplineDto[]>([]);
   const [modalLoading, setModalLoading] = useState(false);
   const [modalData, setModalData] = useState(null as ModalData);
```

Comment:
I would like to avoid typecast here
```suggestion
  const [data, setData] = useState<TaskDto[]>([]);
  const [disciplines, setDisciplines] = useState<DisciplineDto[]>([]);
  const [modalLoading, setModalLoading] = useState(false);
  const [modalData, setModalData] = useState<ModalData>(null);
```